libs/shared/types/src/lib/dto/leaderboard/leaderboard-pagination.dto.ts

```diff
@@ -0,0 +1,9 @@
+import { IsOptional } from 'class-validator';
+
+export class LeaderboardPaginationQueryDto {
+  @IsOptional()
```

Comment:
Wilsch es du jetzt no umbaua bevor mir es mergen? - so das es nur a "page" query gibt, anstatt take und skip

Reply:
so isch ma halt flexibler weils frontend entscheida kann wie viel items uf zmol ma fetched und welche. mir isches glich i kanns o widda umbaua. i persönlich find des die flexiblere variante.

Reply:
jo flexibler isch es, abe sicherheits technisch isch es natürlich o riskanter - so künnt jeder einfach manuell skip 0 take 10000 macha und hät a liste vo allen - I würds guat finda wenn ma MAX_SHOWN_COMPETITORS wenigstens im Backend o überprüfen

Reply:
ma muss sich jo sowieso authorisieren (=igloggt si)? oda wia muansch

Reply:
najo i künnt mi jo trotzdem amelda denn mine tokens extrahiera und manuell über postman an request macha der gültig isch und denn als take 1000000 macha

Reply:
jo und was isches problem dra? Des isch jo erlaubt grundsätzlich

Reply:
? warum hosch den MAX_SHOWN_COMPETITORS im frontend - entweder ma darfs gar net oder im Frontend & Backend isches glich

Reply:
Es goht um des nochladen in chunks im frontend. des isch aussschließlich performance optimierung. der sicherheitsaspekt hot nix mit da pagination zum tua.

Reply:
ah, i hob denkt das du MAX_SHOWN_COMPETITORS gmacht hosch so das der user nur ne bestimmte anzahl an usern in da Rangliste aschoa kann. ok jo, denn künnamas scho so lo, obwohl i net ganz verstoh wie des "performance optimierung" macht

Reply:
asooo jetzt check i was du muansch. ahh na des isch halt einfach dass es nur die top 100 azoagt.

Reply:
mh jo ok, und des us performance technischen gründen? Oder us sicherheits/privatsphäre/datenschutz technischen Gründen (halt so dass da User net alle Benutzernamen usafinda kann)?

Reply:
i hob einfach denkt a limit wär vllt nice. i glob jetzt vastandi o din andra punkt. wenni im frontend 100 limitier sötti im backend o uf 100 limitieren gea?

Reply:
jo genau, des wär glob guat. So künnt o niemand alle Benutzernamen ussafinda

Reply:
fix denn mahi des no